inference-engine/tests/functional/inference_engine/CMakeLists.txt

```diff
@@ -11,19 +11,23 @@ addIeTargetTest(
         INCLUDES
             # TODO: remove after removing `cnn_network_ngraph_imp.hpp`
             ${IE_MAIN_SOURCE_DIR}/src/inference_engine
+            ${OpenVINO_MAIN_SOURCE_DIR}/docs/template_extension
```

Comment:
If you export some symbols from extension library only for tests it is not correct. Because `template_extension` demonstrates how user should create own libraries with custom ops.
And if he use `core.AddExtension("ownLib")`, he shouldn't export operations and extension class from the library.

The code of `template_extension` should be extremely clean without any redundant code.

Reply:
I see. Using make_so_pointer is preferable way to create TemplateExtension::Extension